src/core/server/logging/integration_tests/rolling_file_appender.test.ts

```diff
@@ -0,0 +1,220 @@
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { join } from 'path';
+import { rmdirSync, mkdtempSync, readFileSync, readdirSync } from 'fs';
+import moment from 'moment-timezone';
+import * as kbnTestServer from '../../../test_helpers/kbn_server';
+import { getNextRollingTime } from '../appenders/rolling_file/policies/time_interval/get_next_rolling_time';
+
+const flushDelay = 250;
+const delay = (waitInMs: number) => new Promise((resolve) => setTimeout(resolve, waitInMs));
+const flush = async () => delay(flushDelay);
+
+function createRoot(appenderConfig: any) {
+  return kbnTestServer.createRoot({
+    logging: {
+      silent: true, // set "true" in kbnTestServer
+      appenders: {
+        'rolling-file': appenderConfig,
+      },
+      loggers: [
+        {
+          context: 'test.rolling.file',
+          appenders: ['rolling-file'],
+          level: 'debug',
+        },
+      ],
+    },
+  });
+}
+
+describe('RollingFileAppender', () => {
+  let root: ReturnType<typeof createRoot>;
+  let testDir: string;
+  let logFile: string;
+
+  const getFileContent = (basename: string) =>
+    readFileSync(join(testDir, basename)).toString('utf-8');
+
+  beforeEach(async () => {
+    testDir = mkdtempSync('rolling-test');
+    logFile = join(testDir, 'kibana.log');
+  });
+
+  afterEach(async () => {
+    try {
+      rmdirSync(testDir);
+    } catch (e) {
+      /* trap */
+    }
+    if (root) {
+      await root.shutdown();
+    }
+  });
+
+  const message = (index: number) => `some message of around 40 bytes number ${index}`;
+  const expectedFileContent = (indices: number[]) => indices.map(message).join('\n') + '\n';
+
+  describe('`size-limit` policy with `numeric` strategy', () => {
+    it('rolls the log file in the correct order', async () => {
+      root = createRoot({
+        kind: 'rolling-file',
+        path: logFile,
+        layout: {
+          kind: 'pattern',
+          pattern: '%message',
+        },
+        policy: {
+          kind: 'size-limit',
+          size: '100b',
+        },
+        strategy: {
+          kind: 'numeric',
+          max: 5,
+          pattern: '.%i',
+        },
+      });
+      await root.setup();
+
+      const logger = root.logger.get('test.rolling.file');
+
+      // size = 100b, message.length ~= 40b, should roll every 3 message
```

Comment:
we create a buffer of fixed size to make sure to have better control over the rolling process 
```js
const asciiBuf = Buffer.alloc(5, 'a', 'utf8');
```

Reply:
I'm only using ascii chars in the message, so we kinda know that `sizeInBytes === msg.length`, but I can use a buffer to add more control if we want to.